### PR TITLE
[Fix] Replace mispronounced words in TTS using hack method

### PR DESCRIPTION
**Pull request by 6drf21e**
This PR addresses the issue of mispronounced words in the TTS system by implementing a hack method. The main idea is as follows:
1. Identify correctly pronounced characters by ChatTTS.
2. Replace the mispronounced characters with correctly pronounced ones.

For example:
- Original: 关关雎鸠，在河之洲。窈窕淑女，君子好逑。
- Corrected: 关关**居**鸠，在河之洲。**咬挑**淑女，君子好**求**。

The replacement rule file **ChatTTS/homophones_map.json** contains 16,000 entries.

Rule creation process:
1. Use a word corpus to let ChatTTS infer the text.
2. Record any discrepancies between the inferred text and the input.
3. Find homophones with the correct pronunciation for the mismatched characters.

Corpus used: [Tencent AI Lab Embedding Corpora for Chinese and English Words and Phrases](https://ai.tencent.com/ailab/nlp/en/embedding.html)

Limitations:
1. Some characters and words might not be covered.
2. Some characters do not have correctly pronounced homophones, e.g., “sǒu 叟”.

**中文:**

本次PR通过使用hack办法解决TTS系统中读错、漏读的问题。主要的构思如下：
1. 找到ChatTTS读音正确的字。
2. 将读错的字符替换成读音正确的字。

例如：
- 原文: 关关雎鸠，在河之洲。窈窕淑女，君子好逑。
- 替换后: 关关**居**鸠，在河之洲。**咬挑**淑女，君子好**求**。

替换规则文件 **ChatTTS/homophones_map.json** 包含1.6万条规则。

规则制作流程：
1. 使用词库，让ChatTTS推理文本。
2. 记录推理文本和输入文本之间的不一致。
3. 将不一致的字找到对应的正确读音的同音字。

当前版本所使用的词库：[腾讯AI实验室中英文词语嵌入语料库](https://ai.tencent.com/ailab/nlp/zh/embedding.html)

缺陷：
1. 可能有未覆盖的字词。
2. 有些字找不到能读正确的字，比如“sǒu 叟”。

### CODE REVIEW

File: ChatTTS/utils/infer_utils.py [hunk 1]
```diff
@@ -138,4 +140,15 @@ def apply_half2full_map(text):
...
+    return text.translate(translation_table)
+
+def load_homophones_map():
+    package_dir = os.path.dirname(__file__)
+    mapping_path = os.path.join(package_dir, '..', 'res', 'homophones_map.json')
+    with open(mapping_path, 'r', encoding='utf-8') as f:
+        return json.load(f)
+
+def apply_homophones_map(text):
+    homophones_map = load_homophones_map()
```
Comment (fumiama) [inline ChatTTS/utils/infer_utils.py:152]: 如果每替换一个字就读一次文件，开销太大了。建议将此map缓存起来，参考normalizer的写法。

**Review by fumiama — CHANGES_REQUESTED** [review]

**Comment by fumiama** [comment]
p.s. 建议在代码中标明加载的json的出处。

File: ChatTTS/utils/infer_utils.py (hunk 1 above)
Comment (6drf21e) [reply ChatTTS/utils/infer_utils.py]: 确实是这样，按normalizer的写法修改了。并详细列出替换规则的制作过程和语料库来源。

**Review by fumiama — APPROVED** [review]
Thanks!

**Comment by aliencaocao** [comment]
Im curious on whether this hack has been verified to not affect the output quality by any means. By swapping words, you are scrambling up phrases into meaningless indiv characters. By right this should affect the linguistic ability of the model.

**Comment by fumiama** [comment]
> By right this should affect the linguistic ability of the model.

It is better than no sound if the model cannot recognize that character. We will remove (or shrink) this replace if the model is updated and it can recognize new characters.

**Comment by 6drf21e** [comment]
The replacements only target incorrectly output characters. In modern novels, the replacement rate is low. For example, in "Twenty Thousand Leagues Under the Sea":

- **Total Chinese characters**: 214,526
- **Replaced characters**: 316
- **Replacement rate**: 0.15%

#### Top 10 Replaced Characters:
| Original | Replacement | Count |
|----------|-------------|-------|
| 鲛        | 教          | 43    |
| 鳃        | 塞          | 32    |
| 桅        | 维          | 27    |
| 舷        | 闲          | 22    |
| 颚        | 恶          | 16    |
| 呷        | 嘎          | 12    |
| 锨        | 先          | 12    |
| 獭        | 塔          | 8     |
| 岖        | 区          | 7     |
| 囱        | 聪          | 6     |

In ancient novels, the rate is higher. For example, in "Romance of the Three Kingdoms":

- **Total Chinese characters**: 493,054
- **Replaced characters**: 6,946
- **Replacement rate**: 1.41%

#### Top 10 Replaced Characters:
| Original | Replacement | Count |
|----------|-------------|-------|
| 郃        | 和          | 249   |
| 岱        | 带          | 189   |
| 惇        | 蹲          | 161   |
| 褚        | 楚          | 154   |
| 傕        | 觉          | 136   |
| 汜        | 四          | 134   |
| 讫        | 气          | 133   |
| 赍        | 机          | 119   |
| 綝        | 陈          | 109   |
| 瑁        | 帽          | 103   |

**Pull request merged**